zetaclient/chains/evm/signer/outbound_tracker_reporter.go

```diff
@@ -0,0 +1,86 @@
+// Package signer implements the ChainSigner interface for EVM chains
+package signer
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/zetacore/zetaclient/chains/evm"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/evm/rpc"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	"github.com/zeta-chain/zetacore/zetaclient/logs"
+)
+
+// reportToOutboundTracker reports outboundHash to tracker only when tx receipt is available
+func (signer *Signer) reportToOutboundTracker(
+	ctx context.Context,
+	zetacoreClient interfaces.ZetacoreClient,
+	chainID int64,
+	nonce uint64,
+	outboundHash string,
+	logger zerolog.Logger,
+) {
+	// prepare logger
+	logger = logger.With().
+		Str(logs.FieldMethod, "reportToOutboundTracker").
+		Int64(logs.FieldChain, chainID).
+		Uint64(logs.FieldNonce, nonce).
+		Str(logs.FieldTx, outboundHash).
+		Logger()
+
+	// set being reported flag to avoid duplicate reporting
+	alreadySet := signer.Signer.SetBeingReportedFlag(outboundHash)
+	if alreadySet {
+		logger.Info().
+			Msgf("outbound %s for chain %d nonce %d is being reported", outboundHash, chainID, nonce)
+		return
+	}
+
+	// launch a goroutine to monitor tx confirmation status
+	go func() {
```

Comment:
consider using `bg.Work()`

Reply:
good idea. Is there a way for `bg.Work()` to easily capture the list of parameters: `zetacoreClient`, `chainID`, `nonce`, `outboundHash`?

Reply:
Yes, you can use drop-in replacement:
```go
go func() {
 // ...
})
```

vs 

```go
bg.Work(func(){
  // ...
}, bg.WithLogger(logger), bg.WithName("lorem ipsum"))
```

Reply:
https://github.com/zeta-chain/node/pull/2628/commits/08107c9ab5e91e40bb6d1f754cbe7f9c847f108c